src/Groups/types.jl

```diff
@@ -198,6 +214,68 @@ struct AutomorphismGroup{T} <: GAPGroup
   end
 end
 
+################################################################################
+#
+#  Composite Groups
+#
+################################################################################
+
+
+"""
+    DirectProductOfGroups
+Either direct product of two or more groups of any type, or subgroup of a direct product of groups.
+"""
+struct DirectProductOfGroups <: GAPGroup
+  X::GapObj
+  L::Vector{<:GAPGroup}   # list of groups
+  n::Int64           # length(L)
+  Xfull::GapObj      # direct product of the GAP groups of L
+  isfull::Bool     # true if G is direct product of the groups of L, false if it is a proper subgroup
+
+
+  function DirectProductOfGroups(G::GapObj, L::Vector{<:GAPGroup}, n::Int64, Xf::GapObj, isf::Bool)
+    z = new(G,L,n,Xf,isf)
+    return z
+  end
+end
+
+
+"""
+    SemidirectProductOfGroups{S,T}
+Semidirect product of two groups of type `S` and `T` respectively, or subgroup of a semidirect product of groups.
+"""
+struct SemidirectProductOfGroups{S<:GAPGroup, T<:GAPGroup} <: GAPGroup 
+  X::GapObj
+  N::S              # normal subgroup
+  H::T              # group acting on N
+  f::GAPGroupHomomorphism{T,AutomorphismGroup{S}}
+  Xfull::GapObj         # full semidirect product: X is a subgroup of Xfull. 
+  isfull::Bool     # true if X==Xfull
+
+  function SemidirectProductOfGroups(G::GapObj, N::S, H::T, f, Xf::GapObj, isf::Bool) where S where T
+    z = new{S,T}(G,N,H,f,Xf,isf)
+    return z
+  end
+end
+
+"""
+    WreathProduct
+Wreath product of a group `G` and a group of permutations `H`.
+"""
+struct WreathProduct <: GAPGroup
```

Comment:
This name doesn't fit with the others; I would suggest `WreathProductGroup`.

Reply:
Done.